app/controllers/messages_controller.rb

```diff
@@ -19,8 +19,12 @@ def index
       from_table = from_table.where.not(sender_id: blocked_ids).left_outer_joins(:sender).where('users.deleted IS NULL OR users.deleted = false')
       includes << :sender
     end
+    thread_counts = Message.from(from_table, "messages")
+      .select('thread_id, COUNT(*) as thread_count')
+      .group('thread_id').to_sql
     @messages = Message.from(from_table, "messages").joins(:first_thread)
-      .select('*', 'first_threads_messages.subject as thread_subject')
+      .joins("LEFT JOIN (#{thread_counts}) AS thread_counts ON thread_counts.thread_id = messages.thread_id")
+      .select('*', 'first_threads_messages.subject as thread_subject', 'thread_counts.thread_count')
```

Comment:
What are you doing here? Or -- what is the purpose of this, I suppose.

Reply:
Okay i see the *purpose* further down is for num_in_thread but i'm still moderately confused as to how this even works.

Reply:
This makes the database engine do the counting, producing a new table, and then joins that onto the existing query.

It is, I admit, a bit esoteric.

Reply:
OH it's a SUBQUERY

Reply:
Okay, I... don't think I'd do that that way but i'm not sure how *else* you would do it in one query either, because you don't want it grouped...

...it's also annoying that you can't to_sql rails `count` apparently, that would simply your subquery...